### PR TITLE
Add UI scaffold for account related screens

### PR DESCRIPTION
**Pull request by lukstbit**
### What does it fix?
This PR implements the basic UI scaffolding for account related screens. There are still visual tweaks to be made but those need to be implemented in the theme in another PR. Also, the last screen for resetting the password isn't yet implemented, this will be discussed in the appropriate issue.

### How has it been tested?

There is nothing to test.

### CODE REVIEW

**Comment by alexandru-calinoiu** [comment]
Thank you for your submission, before merging it will be great if you could review https://github.com/code4romania/teacher-workout-android/pull/18

The pr proposes an app arch that should streamline collaboration, but will require some refactoring this pr, mostly moving code around.

Also we where discussing to use [jetpack compose](https://developer.android.com/jetpack/compose) for the UI. We did not do a very good job at communicating this, planning to do a pr to the Readme with the proposed frameworks and patterns.

**Comment by lukstbit** [comment]
I have no problems writing the UI in compose, I wrote it using normal views because compose wasn't mentioned anywhere. Compose is still in beta but with the release of android 12 it will probably get to 1.0.0(or soon after the release) so we will have a stable release.

> We did not do a very good job at communicating this, planning to do a pr to the readme with the proposed frameworks and patterns.

This will be really helpful, to also clarify things like [threading ](https://github.com/code4romania/teacher-workout-android/issues/1#issuecomment-868967123)

**Comment by alexandru-calinoiu** [comment]
> I have no problems writing the UI in compose, I wrote it using normal views because compose wasn't mentioned anywhere. Compose is still in beta but with the release of android 12 it will probably get to 1.0.0(or soon after the release) so we will have a stable release.
> 
> > We did not do a very good job at communicating this, planning to do a pr to the readme with the proposed frameworks and patterns.
> 
> This will be really helpful, to also clarify things like [threading ](https://github.com/code4romania/teacher-workout-android/issues/1#issuecomment-868967123)

I've opened a pr for the readme with some tech https://github.com/code4romania/teacher-workout-android/pull/19

**Comment by lukstbit** [comment]
I've created the basic UI for the account related screens. Because the figma design doesn't have a mockup for showing the outcomes for registration/signing in/resetting password to the user I used a simple panel with information like in the images below. Personally I think a dialog would be more suited.

registration_loading1 registration_failure1 registration_success1

**Comment by AlexandraDamaschin** [comment]
@lukstbit Sorry for the delay, we merged Calin's PR and I've created a new one with some improvements based on your review which hopefully will be merged soon. 

When you get a chance to rebase this PR let us know and will make CR. 

As for the outcomes, I would agree with you for adding dialogs in order to let the user know what happened, but I am not the right person to make the final decision. Maybe @catileptic can help us to decide this or to tell us to who we should speak in order to decide?

**Comment by catileptic** [comment]
@MihaiPantiru, have you, in the iOS app, already implemented these screens? And, if so, have you decided what element to use in order to tell the user whether their actions were successful or failed with an error? I think the most important part is for the iOS and Android apps to be consistent, so if the iOS one already made this decision, let's mirror it. If not, @MihaiPantiru, what is your opinion on the element that we should use?

**Comment by MihaiPantiru** [comment]
@catileptic @vladstanescu94 implemented for iOS and he added the errors like we have on Figma, inline red text below fields and on top red text for general errors.
277E1005-23CF-43AC-B1AB-C754A0C43952

**Comment by lukstbit** [comment]
@AlexandraDamaschin I rebased. Note that I had to disable kapt because it throws some compiler errors. I don't know if it's something  on my side or kapt has issues with dynamic feature modules/compose but I couldn't make the code compile without disabling it.

@MihaiPantiru I implemented the errors as shown in the design mockups. The question was about what to show to the user after clicking to (create an account/signing in or resetting the password) to show loading and the result of the action(like failure to create an account etc). Above, I've attached some images with my current implementation where I disabled the inputs and then displayed a panel inline with the information. I was thinking about using a blocking dialog so the user it's kind of forced to wait to see the outcome of creating an account/ resetting the password etc. I was wondering how the iOS app handles this.

**Comment by AlexandraDamaschin** [comment]
@lukstbit  Thanks, will look over it. @vladstanescu94 Maybe you help us with more information about how you handled the cases after the user clicks to create an account to show the loading/result of the action? (See detailed question :top: )

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/ApplicationLogo.kt
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/landing/ApplicationLogo.kt:12]: The logo can't be extracted as a vector from figma? Or it's still pixeled? I can try to help finding the original one

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt]: We should make this to uppercase as per design

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt:19]: Maybe we can try to round the corners of the button? And add the background color to the screen?

**Comment by lukstbit** [comment]
@AlexandraDamaschin Ok. Is anyone working on the other added UI related issues? If not I can work on them.

File: app/build.gradle.kts
Comment (AlexandraDamaschin) [inline app/build.gradle.kts]: Maybe we can extract this into variables so it will be easier to maintain our code

File: app/src/main/AndroidManifest.xml
Comment (AlexandraDamaschin) [inline app/src/main/AndroidManifest.xml]: Could we add a TODO in order to fix this or don't forget to look for ways to fix it?

File: app/src/main/kotlin/com/teacherworkout/android/theming/Colors.kt
Comment (AlexandraDamaschin) [inline app/src/main/kotlin/com/teacherworkout/android/theming/Colors.kt:15]: I will ask more about this and come back with details after this Saturday.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt:30]: Why do you name the functions with CamelCase?  Android studio seem to be pretty upset about that
image

File: features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt
Comment (AlexandraDamaschin) [inline features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt]: It would help a lot to refactor all these dimensions into one file and reuse them from there. 
https://developer.android.com/jetpack/compose/resources#dimensions

**Review by AlexandraDamaschin — CHANGES_REQUESTED** [review]
Some improvements which hopefully will make out life easier. Let me know if I can help with something.

**Comment by AlexandraDamaschin** [comment]
> @AlexandraDamaschin Ok. Is anyone working on the other added UI related issues? If not I can work on them.

Yeah, me and one colleague are working on the onboarding steps, except from that nothing I am aware of. We are having a meeting this Saturday with people coding on, so we will work on the issues which don't have any comments.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/ApplicationLogo.kt
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/ApplicationLogo.kt:12]: I'm sure it is available in Figma, it's just that I don't have an account and I don't want to make one either :) . So I made a simple screenshot and use that as a placeholder. The vector logo image can simple replace the png logo without any code changes.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt]: Yes, nice catch!

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt:19]: I rounded the button. I'm not sure I understand the color request, as right now the implementation matches the figma design.

File: app/build.gradle.kts
Comment (lukstbit) [reply app/build.gradle.kts]: Done, also added some documentation info.

File: app/src/main/AndroidManifest.xml
Comment (lukstbit) [reply app/src/main/AndroidManifest.xml]: I deleted the entry. The MainActivity will be added in one of the features modules, most likely home.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt:30]: My IDE(latest rc release) doesn't show that warning. I used PascalCase because that is the recommendation in the [compose style guide](https://github.com/androidx/androidx/blob/androidx-main/compose/docs/compose-api-guidelines.md).

File: features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt]: Yes, I should've done this from the start!

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/ApplicationLogo.kt
Comment (AlexandraDamaschin) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/ApplicationLogo.kt:12]: All right, then I will take a note for myself to make that in another PR. Thanks for explaining.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt
Comment (AlexandraDamaschin) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt:19]: image
This is the background color about which I was asking.

File: features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt
Comment (AlexandraDamaschin) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/auth/AuthenticateScreen.kt:30]: My bad, I didn't had the latest version so that makes total sense. 

As a side joke, we had loads of fun today with my colleagues based on if this is CamelCase or PascalCase. ( IMO it's also CamelCase) :)) Hope you enjoy the joke :))

File: features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt
Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt:19]: I see it now. I'm using my monitor with night light which dims the white so there was no chance of my noticing the color differences.

Comment (AlexandraDamaschin) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt:19]: Will you get a chance to do that or I can help with this in another PR. 
Was thinking to merge this PR tonight.

Comment (lukstbit) [reply features/account/src/main/kotlin/com/teacherworkout/features/account/landing/RegistrationButton.kt:19]: @AlexandraDamaschin I pushed the requested changes including the background color!

**Review by AlexandraDamaschin — APPROVED** [review]
Does this improve software quality? Yes

**Pull request merged**
